tests/test_api.py

```
@@ -1,288 +1,213 @@
+    def test_post_http_error(self, client, mock_response):
+        response = mock_response(400, "Bad Request")
+        response.raise_for_status.side_effect = requests.exceptions.HTTPError(
+            "400 error"
         )
-        mock_post.return_value = mock_response
-
-        with patch("src.lib.api.api_logger") as mock_logger:
-            response = self.client.post("/test", data={})
-
-            assert response == mock_response
+        with patch("requests.post", return_value=response) as mock_post, patch(
```

Ruff (0.12.2)

Local variable `mock_post` is assigned to but never used

Remove assignment to unused variable `mock_post`

(F841)